lib/wootric/test.js

```diff
@@ -0,0 +1,155 @@
+
+var Analytics = require('analytics.js').constructor;
+var integration = require('analytics.js-integration');
+var plugin = require('./');
+var tester = require('analytics.js-integration-tester');
+var sandbox = require('clear-env');
+
+describe('Wootric', function(){
+  var Wootric = plugin;
+  var wootric;
+  var analytics;
+  var options = {
+    account_token: 'NPS-01fe3cbc'
+  };
+
+  beforeEach(function(){
+    analytics = new Analytics;
+    wootric = new Wootric(options);
+    analytics.use(plugin);
+    analytics.use(tester);
+    analytics.add(wootric);
+    //setup Wootric to survey immediately
+    window.wootric_survey_immediately = true;
+  });
+
+  afterEach(function(){
+    analytics.restore();
+    analytics.reset();
+    wootric.reset();
+    sandbox();
+  });
+
+  it('should have the right settings', function(){
+    analytics.compare(Wootric, integration('Wootric')
+      .assumesPageview()
+        .option('account_token', '')
+        .global('wootricSettings')
+        .global('wootric_survey_immediately')
+        .global('wootric')
+    )
+  });
+
+  describe('before loading', function(){
+    beforeEach(function(){
+      analytics.stub(wootric, 'load');
+    });
+
+    it('should not have a wootric object', function(){
+      analytics.assert(!window.wootric);
+    })
+
+    describe('#initialize', function(){
+      beforeEach(function(){
+        analytics.initialize();
+        analytics.page();
+      });
+
+      it ('should setup the wootricSettings object', function(){
+        analytics.assert(window.wootricSettings);
+        analytics.assert(window.wootricSettings instanceof Object)
```

Comment:
What you want here is actually to test if this is a plain object; `instanceof` object is pretty permissive in JavaScript. For example, all of these evaluate to `true` when it's likely not what you mean:

```js
(function(){}) instanceof Object
[] instanceof Object
new String('abc') instanceof Object
```

Replacement:

```js
var is = require('is');
is.object(window.wootricSettings);
```